pkg/operator/ceph/object/controller.go

```diff
@@ -254,9 +253,6 @@ func (r *ReconcileCephObjectStore) reconcile(request reconcile.Request) (reconci
 			if err != nil {
 				return reconcile.Result{}, errors.Wrapf(err, "failed to delete store %q", cephObjectStore.Name)
 			}
```

Comment:
I think we should move this block https://github.com/rook/rook/pull/6208/files#diff-7e97b37c53b4739ef8382f7b481ab8dcR240-R244 after `cfg.deleteStore()`, if `cfg.deleteStore()` fails we want to retry again, if we close the channel and remove from the map before then we will never re-enter this block.

Reply:
Thanks for the suggestion, Done.

Reply:
Did you validate again with this new change?

Reply:
> Did you validate again with this new change?
Yes, I validate the latest changes.